src/lib/gadgets/gadgets.ts

```diff
@@ -474,6 +474,45 @@ const Gadgets = {
     div(x: Field3, y: Field3, f: bigint) {
       return ForeignField.div(x, y, f);
     },
+
+    /**
+     * Optimized multiplication of sums in a foreign field, for example: `(x - y)*z = a + b + c mod f`
+     *
+     * Note: This is much more efficient than using {@link ForeignField.add} and {@link ForeignField.sub} separately to
+     * compute the multiplication inputs and outputs, and then using {@link ForeignField.mul} to constrain the result.
+     *
+     * The sums passed into this gadgets are "lazy sums" created with {@link ForeignField.Sum}.
+     * You can also pass in plain {@link Field3} elements.
+     *
+     * **Assumptions**: The assumptions on the _summands_ are analogous to the assumptions described in {@link ForeignField.mul}:
+     * - each summand's limbs are in the range [0, 2^88)
+     * - summands that are part of a multiplication input satisfy `x[2] <= f[2]`
```

Comment:
Is this an assumption or constrained somewhere?

Reply:
assumption on the inputs

Reply:
(needs to be constrained by the caller)